src/mapbox/mapbox.js

```diff
@@ -250,6 +254,14 @@ export default class Mapbox {
 
     this._create(props);
 
+    // Hijack dimension properties
+    // This eliminates the timing issue between calling resize() and DOM update
+    const {container} = props;
+    container.__defineGetter__('offsetWidth', () => this.width);
+    container.__defineGetter__('clientWidth', () => this.width);
+    container.__defineGetter__('offsetHeight', () => this.height);
+    container.__defineGetter__('clientHeight', () => this.height);
```

Comment:
Clever. 

Would Object.defineProperty be better here? Sounds like defineGetter is deprecated.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/__defineGetter__